### PR TITLE
Post table

### PR DESCRIPTION
**Pull request by kelvin8773**
* restructure component folder
* add Post table component
* finish style and function of PostTable component

### CODE REVIEW

File: src/main/components/search/heatMap.js [hunk 1]
```diff
@@ -75,18 +77,19 @@ const HeatMapCell = Styled.td`
...
 
-
 const HeatMap = () => {
   const [clickedIndex, setClickedIndex] = useState(null);
   const posts = useSelector((state) => state.posts);
   const localTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
-
   const getIndex = (weekday, hour) => weekday * 100 + hour;
+  const selectedPosts = clickedIndex
+    ? posts[Math.floor(clickedIndex / 100)][clickedIndex % 100] : [];
```
Comment (jkettmann) [inline src/main/components/search/heatMap.js:92]: I'd extract that into a function as well. Something like `getDay` and `getHour`. It took me again a close look to understand what's happening.

Maybe even extract all this index logic into one file so it's all in one place. Imagine you were new to the code base and would see all this `* 100`, `/ 100`, `% 100`. It looks confusing. If you would only see `getIndex`, `getDay`, `getHour` that's very easy to understand

File: src/main/components/search/postTable.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import Styled from 'styled-components';
+import PropTypes from 'prop-types';
+import dayjs from 'dayjs';
+import { REDDIT_USER_BASE_URL } from '../../../config/constants';
+
+const TableTitle = Styled.h2`
+  font-family: 'Bitter', serif;
+  font-size: 24px;
+  color: ${({ theme }) => theme.colors.black};
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:10]: Should this maybe be defined globally for h2?

File: src/main/components/search/postTable.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+`;
+
+const TableWrapper = Styled.table`
+  box-sizing: border-box;
+  width: 786px;
+  margin-top: 12px;
+  border: solid 1px #dddddd;
+  font-size: 14px;
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:18]: This font-size may be a constant in the theme

File: src/main/components/search/postTable.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+`;
+
+const TableHeader = Styled.th`
+  box-sizing: border-box;
+  height: 35px;
+  font-size: 14px;
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:24]: This font-size may be a constant in the theme

File: src/main/components/search/postTable.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  font-weight: 600;
+  border: solid 1px #dddddd;
+  color: ${({ theme }) => theme.colors.black};
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:27]: Do we need to set the color here? What is it's global value?

File: src/main/components/search/postTable.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+const TableData = Styled.td`
+  box-sizing: border-box;
+  height: 34px;
+  font-size: 14px;
+  font-weight: 500;
+  border: solid 1px #dddddd;
+  padding-left: 12px;
+  color: ${({ theme }) => theme.colors.black};
+  vertical-align: middle !important;
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:43]: Oh oh! :smile: `!important` is never a good sign. Why is this necessary

File: src/main/components/search/postTable.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  box-sizing: border-box;
+  height: 35px;
+  font-size: 14px;
+  font-weight: 600;
+  border: solid 1px #dddddd;
+  color: ${({ theme }) => theme.colors.black};
+  text-align: start;
+  padding-left: 12px;
+  padding-right: 10px;
+  vertical-align: middle !important;
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:31]: Oh oh! :smile: `!important` is never a good sign. Why is this necessary

File: src/main/components/search/postTable.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        </thead>
+        <tbody>
+          {
+            sortedPosts.map((post) => (
+              <tr key={post.id}>
+                <TableData>
+                  <a
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    href={post.full_link}
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:91]: The name `full_link`, `created_utc`, and `num_comments` is inconsistent. I would stick to camelCase for all names

Comment (jkettmann) [reply src/main/components/search/postTable.js:91]: Except for constants :smile:

Comment (jkettmann) [reply src/main/components/search/postTable.js:91]: Or are these the names you get from the API directly?

File: src/main/components/search/postTable.js (hunk 6 above)
Comment (kelvin8773) [reply src/main/components/search/postTable.js:43]: ok, I actually don't need that, not sure why previous don't work.

File: src/main/components/search/postTable.js (hunk 7 above)
Comment (kelvin8773) [reply src/main/components/search/postTable.js:31]: removed.

File: src/main/components/search/postTable.js (hunk 8 above)
Comment (kelvin8773) [reply src/main/components/search/postTable.js:91]: yes, it is from API data, I couldn't change it, unfortunately.

File: src/main/components/search/heatMap.js [hunk 9]
```diff
@@ -87,9 +93,8 @@ const HeatMap = () => {
   const [clickedIndex, setClickedIndex] = useState(null);
   const posts = useSelector((state) => state.posts);
   const localTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
-  const getIndex = (weekday, hour) => weekday * 100 + hour;
   const selectedPosts = clickedIndex
```
Comment (jkettmann) [inline src/main/components/search/heatMap.js:96]: Btw: I think here is a bug. What if clickedIndex is `0`? That would be the very first cell in the heatmap, right? So I tested it and the first cell of the heatmap never shows the table

Comment (kelvin8773) [reply src/main/components/search/heatMap.js]: it cause by this line of code, it is fixed now.
```Javascript
 const selectedPosts = clickedIndex !== null
    ? posts[getDay(clickedIndex)][getHour(clickedIndex)] : [];
```

File: src/config/theme/global.js [hunk 10]
```diff
@@ -10,14 +10,13 @@ const GlobalStyle = createGlobalStyle`
     font-style: normal;
     line-height: normal;
     letter-spacing: normal;
-    font-size: 16px;
-    color: ${({ theme }) => theme.colors.grayBase};
+    font-size: 14px;
```
Comment (jkettmann) [inline src/config/theme/global.js:13]: 16px was correct here. Now the font-size on the home page doesn't fit. Sorry, maybe I confused you. I just meant that 14px should be a constant in the theme, not the global styles :wink:

Comment (jkettmann) [reply src/config/theme/global.js:13]: Like `fontSizeSmall` or so

File: src/config/theme/global.js [hunk 11]
```diff
@@ -10,14 +10,13 @@ const GlobalStyle = createGlobalStyle`
...
+    color: ${({ theme }) => theme.colors.black};
```
Comment (jkettmann) [inline src/config/theme/global.js:14]: And sorry, the color was actually correct. My bad

File: src/main/components/search/postTable.js (hunk 5 above)
Comment (jkettmann) [reply src/main/components/search/postTable.js]: Yes we need to set it :smile: Sorry for the confusion

File: src/main/components/search/postTable.js (hunk 8 above)
Comment (jkettmann) [reply src/main/components/search/postTable.js:82]: You can map the response data in the [reducer](https://github.com/ooloo-io/reddit-timer-kelvin8773/blob/postTable/src/slices/postsSlice.js#L8). Assuming that `payload` is an array you can simply run something like.

```
payload.map(post => ({
  url: post.full_link,
  ...
}))
``` 

This way you can give them more consistent and meaningful names for your app

**Comment by kelvin8773** [comment]
No problem, Johannes

File: src/main/components/search/postTable.js [hunk 12]
```diff
@@ -9,6 +9,8 @@ const TableWrapper = Styled.table`
   width: 786px;
   margin: 12px auto 0;
   border: solid 1px #dddddd;
+  Font-size: ${({ theme }) => theme.fonts.small};
```
Comment (jkettmann) [inline src/main/components/search/postTable.js:12]: This seems to work but I'd still consider it a typo. `font-size` should be all lower-case

Comment (kelvin8773) [reply src/main/components/search/postTable.js]: Thanks, you are really aiming to detail!
